server/controllers/lettersControllers/lettersControllers.js

```diff
@@ -22,9 +41,10 @@ exports.getAllLetters = async (req, res) => {
 
 exports.getLettersById = async (req, res) => {
     try {
-
+        const letterId = req.query.id
```

Comment:
This line should be `rq.params.id` instead of `req.query.id`. We are grabbing the value from a parameter not a query.